econml/ortho_forest.py

```diff
@@ -493,20 +439,27 @@ def nuisance_estimator(Y, T, X, W, sample_weight=None, split_indices=None):
                 kfold_it = KFold(n_splits=2, shuffle=True, random_state=this_random_state).split(X)
                 split_indices = list(kfold_it)[0]
             if W is not None:
-                try:
-                    T_hat = _cross_fit(model_T, W, T, split_indices, sample_weight=sample_weight)
-                    Y_hat = _cross_fit(model_Y, W, Y, split_indices, sample_weight=sample_weight)
-                except ValueError as exc:
-                    raise ValueError("The original error: {0}".format(str(exc)) +
-                                     " This might be caused by too few sample in the tree leafs." +
-                                     " Try increasing the min_leaf_size.")
-                return Y_hat, T_hat
+                X_tilde = np.concatenate((X, W), axis=1)
             else:
-                return np.zeros(Y.shape), np.zeros(T.shape)
+                X_tilde = X
+
+            try:
+                if second_stage:
+                    T_hat = _cross_fit(model_T, X_tilde, T, split_indices, sample_weight=sample_weight)
+                    Y_hat = _cross_fit(model_Y, X_tilde, Y, split_indices, sample_weight=sample_weight)
+                else:
+                    T_hat = model_T.fit(X_tilde, T, sample_weight=sample_weight).predict(X_tilde)
```

Comment:
The weights are never used in the first stage. I think we should omit passing them in here so the user doesn't have to pass in a weighted model for `model_T` and `model_Y`. This will potentially avoid some issues in the future. Same for the `DiscreteTreatmentOrthoForest` nuisance estimator.

Reply:
check out if I did it correctly

Reply:
@moprescu Looks like I need to dismiss this review to enable checkin - I assume that's okay with you.